### PR TITLE
Add unit tests for CronJob controller watch CronJobs

### PR DESCRIPTION
**Pull request by pacoxu**
/kind cleanup

Before
```
ok      k8s.io/kubernetes/pkg/controller/cronjob        0.908s  coverage: 52.5% of statements
```
After
```
ok      k8s.io/kubernetes/pkg/controller/cronjob        1.009s  coverage: 60.2% of statements
```
Which issue(s) this PR fixes:
xref https://github.com/kubernetes/enhancements/issues/3140

Special notes for your reviewer:
CronJobTimeZone will GA in v1.27 and this helps to adding the code coverage of it

```release-note
None
```

### CODE REVIEW

**Comment by sftim** [comment]
/retitle Add unit tests for CronJob controller watch CronJobs

File: pkg/controller/cronjob/cronjob_controllerv2_test.go [hunk 1]
```diff
@@ -1674,3 +1700,46 @@ func TestControllerV2GetJobsToBeReconciled(t *testing.T) {
...
+	manager, sharedInformerFactory := newControllerFromClient(clientset, controller.NoResyncPeriodFunc)
+	manager.cronJobListerSynced = alwaysReady
+	manager.jobListerSynced = alwaysReady
+
+	var testCronJob batchv1.CronJob
+	received := make(chan struct{})
+
+	// The update sent through the fakeWatcher should make its way into the workqueue,
+	// and eventually into the syncHandler.
+	manager.syncHandler = func(ctx context.Context, cronJob *batchv1.CronJob, _ []*batchv1.Job) (*batchv1.CronJob, *time.Duration, bool, error) {
```
Comment (soltysh) [inline pkg/controller/cronjob/cronjob_controllerv2_test.go:1717]: I'm not sure I follow, why do you want to test watches? That's already handled by the tests in client-go, no?

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/controller/cronjob/cronjob_controllerv2_test.go (hunk 1 above)
Comment (pacoxu) [reply pkg/controller/cronjob/cronjob_controllerv2_test.go:1717]: This should be covered by the e2e test for cronjob. This UT just confirmed that the `cronjob_controller` can be running and syncing.

I found that `func (jm *ControllerV2) Run(ctx context.Context, workers int)` is not covered in UT and just adding one simple `TestWatch` here.


I follow the style of `TestWatchJobs `. 
https://github.com/kubernetes/kubernetes/blob/4dd887797fe53816faf3cd4d65d600a606714f5d/pkg/controller/job/job_controller_test.go#L3618

Comment (pacoxu) [reply pkg/controller/cronjob/cronjob_controllerv2_test.go:1717]: /priority backlog

Comment (soltysh) [reply pkg/controller/cronjob/cronjob_controllerv2_test.go:1714]: We don't have to cover every possible case in units, if you care about such a test, I'd suggest an integration, but I'm pretty confident we already cover this case. On top of that overwriting `syncHandler` isn't something I'm willing to accept.

Comment (pacoxu) [reply pkg/controller/cronjob/cronjob_controllerv2_test.go:1714]: /close

per comments above.

**Pull request closed without merging**